app/extensions/brave/content/scripts/requestHandler.js

```diff
@@ -0,0 +1,373 @@
+    if (userReg.test(str)) {
+      const elements = str.split('/')
+      return requestHandlerApi.stripParameters(elements.pop())
+    }
+
+    // attribution_link
+    const attrReg = /\/attribution_link\?.*v%3D([^%&]*)(%26|&|$)/
+    if (attrReg.test(str)) {
+      return str.match(attrReg)[1]
+    }
+  },
+
+  stripParameters: (str) => {
+    if (str == null) {
+      return ''
+    }
+
+    // Split parameters
+    if (str.includes('?')) {
+      return str.split('?')[0]
+    }
+
+    // Split folder separator
+    if (str.includes('/')) {
+      return str.split('/')[0]
+    }
+
+    return str
+  },
+
+  // https://github.com/kellym/smartquotesjs
+  getReplacements: () => {
+    return [
+      // triple prime
+      [/'''/g, retainLength => '\u2034' + (retainLength ? '\u2063\u2063' : '')],
+      // beginning "
+      [/(\W|^)"(\w)/g, '$1\u201c$2'],
+      // ending "
+      [/(\u201c[^"]*)"([^"]*$|[^\u201c"]*\u201c)/g, '$1\u201d$2'],
+      // remaining " at end of word
+      [/([^0-9])"/g, '$1\u201d'],
+      // double prime as two single quotes
+      [/''/g, retainLength => '\u2033' + (retainLength ? '\u2063' : '')],
+      // beginning '
+      [/(\W|^)'(\S)/g, '$1\u2018$2'],
+      // conjunction's possession
+      [/([a-z])'([a-z])/ig, '$1\u2019$2'],
+      // abbrev. years like '93
+      [/(\u2018)([0-9]{2}[^\u2019]*)(\u2018([^0-9]|$)|$|\u2019[a-z])/ig, '\u2019$2$3'],
+      // ending '
+      [/((\u2018[^']*)|[a-z])'([^0-9]|$)/ig, '$1\u2019$3'],
+      // backwards apostrophe
+      [/(\B|^)\u2018(?=([^\u2018\u2019]*\u2019\b)*([^\u2018\u2019]*\B\W[\u2018\u2019]\b|[^\u2018\u2019]*$))/ig, '$1\u2019'],
+      // double prime
+      [/"/g, '\u2033'],
+      // prime
+      [/'/g, '\u2032']
+    ]
+  },
+
+  smartQuotes: (str) => {
+    const replacements = requestHandlerApi.getReplacements()
+    if (!replacements || !str) {
+      return ''
+    }
+
+    replacements.forEach(replace => {
+      const replacement = typeof replace[1] === 'function' ? replace[1]({}) : replace[1]
+      str = str.replace(replace[0], replacement)
+    })
+
+    return str
+  },
+
+  removeByPrefix: (str = '') => {
+    if (str == null) {
+      return ''
+    }
+
+    return str.replace(/^[\s\n]*by|@[\s\n]*/i, '').trim()
+  },
+
+  createTitle: (str = '') => {
+    if (str == null) {
+      return ''
+    }
+
+    str = str.trim().replace(/\s{2,}/g, ' ')
+    return requestHandlerApi.smartQuotes(str)
+  },
+
+  // https://github.com/sindresorhus/is-absolute-url
+  isAbsoluteUrl: (url) => {
+    if (!requestHandlerApi.isString(url)) {
+      return
+    }
+
+    return /^[a-z][a-z0-9+.-]*:/.test(url)
+  },
+
+  resolveUrl: (baseUrl, relativePath) => {
+    let url = baseUrl
+
+    if (!relativePath) {
+      return url
+    }
+
+    try {
+      url = new URL(relativePath, [baseUrl])
+    } catch (e) {}
+
+    return url
+  },
+
+  isString: (str) => {
+    return typeof str === 'string'
+  }
+}
+
+module.exports = requestHandlerApi
```

Comment:
should this line check if `module` is defined first? (since content script pages don't use node/npm)

Reply:
done